### PR TITLE
CppCheck InternalHeatGains

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes programming style
 - Use CppCheck as guide
 
**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/InternalHeatGains.cc [hunk 1]
```diff
@@ -367,8 +365,6 @@ namespace InternalHeatGains {
                                                       format("Schedule=\"{}\". Minimum is [{:.1R}]. Values must be >= 0.0.", IHGAlphas(3), SchMin));
                                     ErrorsFound = true;
                                 }
-                            }
-                            if (Item1 == 1) {
```
Comment (rraustad) [inline src/EnergyPlus/InternalHeatGains.cc:371]: [src/EnergyPlus/InternalHeatGains.cc:371]:(style),[duplicateCondition],The if condition is the same as the previous if condition

File: src/EnergyPlus/InternalHeatGains.cc [hunk 2]
```diff
@@ -1096,35 +1090,35 @@ namespace InternalHeatGains {
                 }
             }
 
-            for (int peopleNum = 1; peopleNum <= state.dataHeatBal->TotPeople; ++peopleNum) {
+            for (int peopleNum2 = 1; peopleNum2 <= state.dataHeatBal->TotPeople; ++peopleNum2) {
```
Comment (rraustad) [inline src/EnergyPlus/InternalHeatGains.cc:1093]: Lots of these:

[src/EnergyPlus/InternalHeatGains.cc:1099]:(style),[shadowVariable],Local variable 'peopleNum' shadows outer variable

File: src/EnergyPlus/InternalHeatGains.cc [hunk 3]
```diff
@@ -1530,70 +1522,74 @@ namespace InternalHeatGains {
...
+                    state, state.dataOutRptPredefined->pdchInLtPower, liteName, state.dataHeatBal->Lights(lightsNum2).DesignLevel * mult);
                 PreDefTableEntry(
-                    state, state.dataOutRptPredefined->pdchInLtRetAir, liteName, state.dataHeatBal->Lights(lightsNum).FractionReturnAir, 4);
+                    state, state.dataOutRptPredefined->pdchInLtEndUse, liteName, state.dataHeatBal->Lights(lightsNum2).EndUseSubcategory);
+                PreDefTableEntry(state,
+                                 state.dataOutRptPredefined->pdchInLtSchd,
+                                 liteName,
+                                 GetScheduleName(state, state.dataHeatBal->Lights(lightsNum2).SchedPtr));
+                PreDefTableEntry(
+                    state, state.dataOutRptPredefined->pdchInLtRetAir, liteName, state.dataHeatBal->Lights(lightsNum2).FractionReturnAir, 4);
```
Comment (rraustad) [inline src/EnergyPlus/InternalHeatGains.cc:1592]: This does look strange but it's a result of changing lightNum to lightNum2. Diffs should show if this is good.

File: src/EnergyPlus/Material.cc [hunk 4]
```diff
@@ -1216,7 +1216,7 @@ void GetMaterialData(EnergyPlusData &state, bool &ErrorsFound) // set to true if
         matGas->ROnly = true;
 
         for (NumGas = 0; NumGas < NumGases; ++NumGas) {
-            GasType gasType = matGas->gases[NumGas].type;
+            gasType = matGas->gases[NumGas].type;
```
Comment (rraustad) [inline src/EnergyPlus/Material.cc:1219]: [src/EnergyPlus/Material.cc:1219]:(style),[shadowVariable],Local variable 'gasType' shadows outer variable
[

File: src/EnergyPlus/InternalHeatGains.cc (hunk 3 above)
Comment (Myoldmopar) [reply src/EnergyPlus/InternalHeatGains.cc:1592]: The lack of diffs shows this is good then.

**Comment by Myoldmopar** [comment]
All clean, and runs fine locally.  I'm a little nervous about dropping anything else in, but this is clean and simple.  Thanks @rraustad

**Pull request merged**